src/backup.ts

```diff
@@ -0,0 +1,385 @@
+    | EnumKey<typeof databaseAdmin.spanner.admin.database.v1.Backup.State>
+    | undefined
+  > {
+    const [backupInfo] = await this.getMetadata();
+    const state = backupInfo.state;
+    return state === null || state === undefined ? undefined : state;
+  }
+
+  /**
+   * Retrieves the expiry time of the backup.
+   *
+   * @see {@link #updateExpireTime}
+   * @see {@link #getMetadata}
+   *
+   * @method Backup#getExpireTime
+   * @returns {Promise<PreciseDate | undefined>}
+   *     when resolved, contains the current expire time of the backup if it exists,
+   *     or undefined if the backup does not exist.
+   *
+   * @example
+   * const {Spanner} = require('@google-cloud/spanner');
+   * const spanner = new Spanner();
+   * const instance = spanner.instance('my-instance');
+   * const backup = instance.backup('my-backup');
+   * const expireTime = await backup.getExpireTime();
+   * console.log(`Backup ${backup.formattedName_} expires on ${expireTime.toISOString()}`);
+   */
+  async getExpireTime(): Promise<PreciseDate | undefined> {
+    const [backupInfo] = await this.getMetadata();
+    const expireTime = backupInfo.expireTime;
+    return expireTime ? new PreciseDate(expireTime as DateStruct) : undefined;
+  }
+
+  /**
+   * Checks whether the backup exists.
+   *
+   * @see {@link #getMetadata}
+   *
+   * @method Backup#exists
+   * @returns {Promise<boolean>} when resolved, contains true if the backup
+   *     exists and false if it does not exist.
+   *
+   * @example
+   * const {Spanner} = require('@google-cloud/spanner');
+   * const spanner = new Spanner();
+   * const instance = spanner.instance('my-instance');
+   * const database = spanner.database('my-database');
+   * const backup = instance.backup('my-backup');
+   * const alreadyExists = await backup.exists();
+   * console.log(`Does backup exist? ${alreadyExists}`);
+   */
+  async exists(): Promise<boolean> {
+    try {
+      // Attempt to read metadata to determine whether backup exists
+      await this.getMetadata();
+      // Found therefore it exists
+      return true;
+    } catch (err) {
+      if (err.code === status.NOT_FOUND) {
+        return false;
+      }
+      // Some other error occurred, rethrow
+      throw err;
+    }
+  }
+
+  updateExpireTime(expireTime: PreciseDate): Promise<Backup>;
+  updateExpireTime(
+    expireTime: PreciseDate,
+    callback: UpdateExpireTimeCallback
+  ): void;
+  /**
+   * Sets the expiry time of a backup.
+   *
+   * @see {@link #getExpireTime}
+   *
+   * @method Backup#updateExpireTime
+   * @returns {Promise<Backup>} when resolved, the backup's expire time will
+   *     have been updated.
+   *
+   * @example
+   * const {Spanner} = require('@google-cloud/spanner');
+   * const spanner = new Spanner();
+   * const instance = spanner.instance('my-instance');
+   * const myBackup = instance.backup('my-backup');
+   * const newExpireTime = new PreciseDate(Date.now() + 1000 * 60 * 60 * 24);
```

Comment:
Optional: it would be nice to have `const` for `1000 * 60 * 60 * 24`.

Reply:
Thanks. I made it:

```
const oneDay = 1000 * 60 * 24;
const newExpireTime = new PreciseDate(Date.now() + oneDay)
```